packages/@o3r/eslint-plugin/src/index.ts

```diff
@@ -3,21 +3,25 @@ import noFolderImportForModule from './rules/typescript/no-folder-import-for-mod
 import o3rWidgetTags from './rules/typescript/o3r-widget-tags/o3r-widget-tags';
 import templateAsyncNumberLimitation from './rules/template/template-async-number-limitation/template-async-number-limitation';
 import jsonDependencyVersionsHarmonize from './rules/json/json-dependency-versions-harmonize/json-dependency-versions-harmonize';
+import matchingConfigurationName from './rules/typescript/matching-configuration-name/matching-configuration-name';
 
 module.exports = {
   rules: {
     'no-folder-import-for-module': noFolderImportForModule,
     'template-async-number-limitation': templateAsyncNumberLimitation,
     'o3r-widget-tags': o3rWidgetTags,
-    'json-dependency-versions-harmonize': jsonDependencyVersionsHarmonize
+    'json-dependency-versions-harmonize': jsonDependencyVersionsHarmonize,
+    'matching-configuration-name': matchingConfigurationName
   },
   configs: {
     '@o3r/no-folder-import-for-module': 'error',
     '@o3r/json-dependency-versions-harmonize': 'error',
     '@o3r/template-async-number-limitation': 'warn',
+    '@o3r/matching-configuration-name': 'error',
```

Comment:
because it is actually not required by the framework itself, I think it would make more sense to be `warn` per default

Reply:
It's not required by the framework but if there is a misalignment, overrides of config simply doesn't work.
If we don't make it a strong blocker, we get the issue at runtime and it could have been prevented already at build time.

Reply:
As discussed will be set as `warn` for v9
and to be discussed for v10 if we turn it into `error` or not ?